proxy/proxy.go

```diff
@@ -234,6 +267,65 @@ func (proxy *Proxy) Start(ctx context.Context) error {
 		}
 	}()
 
+	// Start gRPC TLS server
+	grpcTlsExit := make(chan error, 1)
+	proxy.logger.WithFields(logrus.Fields{
+		"address": proxy.GetGrpcTlsAddress(),
+		"tls":     "true",
+	}).Debug("serving grpc")
+	waitGroup.Add(1)
+	go func() {
+		defer waitGroup.Done()
+
```

Comment:
The grpcTLS server seems like a dupe of grpc Server above. I would at least want to see a separate function for the deep nesting of goroutines like `go start(server)`)

Reply:
I think the anonymous function at line 292 should at least be a separate function as right now it handles ~a lot of different tasks. 

`go handleShutdown(proxy.grpcServer)` seems easy enough. (may add a logger field so that the logger outputs "server_type: grpcTls" or "server_type: grpc" to distinguish the logs)

Reply:
Actually I was able to refactor this cleanly.